src/nnet3/nnet-descriptor.h

```diff
@@ -192,6 +198,12 @@ class SwitchingForwardingDescriptor: public ForwardingDescriptor {
   SwitchingForwardingDescriptor(std::vector<ForwardingDescriptor*> &src):
       src_(src) { }
   virtual ~SwitchingForwardingDescriptor() { DeletePointers(&src_); }
+
+  // defined in nnet-utils.h
+  friend void ExtractDescriptorValuesFromForwardingDescriptor(const Nnet &Nnet,
```

Comment:
You say the function extracts values but the variable being modified is called offsets. Consistency is required.